### PR TITLE
Add keras imdb example

### PR DESCRIPTION
**Pull request by hkvision**
## What changes were proposed in this pull request?

Add keras imdb example.
Refactor load_keras API.
Update docs.

## How was this patch tested?

manual run the tests and jenkins

### CODE REVIEW

File: pyspark/bigdl/examples/keras/keras_utils.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+def save_keras_model(keras_model, path):
+    """
+    Save a Keras model to JSON with given path
+    """
+    model_json = keras_model.to_json()
+    with open(path, "w") as json_file:
+        json_file.write(model_json)
+
+
+def parallelize_data(sc, x, y):
```
Comment (zhichao-li) [inline pyspark/bigdl/examples/keras/keras_utils.py]: we have similar method in `common.py`

File: pyspark/bigdl/examples/keras/keras_utils.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from bigdl.util.common import *
+
+
+def save_keras_model(keras_model, path):
```
Comment (zhichao-li) [inline pyspark/bigdl/examples/keras/keras_utils.py]: Save `parameters` as well?

File: pyspark/bigdl/examples/keras/keras_utils.py (hunk 1 above)
Comment (hkvision) [reply pyspark/bigdl/examples/keras/keras_utils.py]: Oh, yes. I find it!

**Comment by hkvision** [comment]
Removed air pollution example for now and may add it after further test @jason-dai

**Comment by zhichao-li** [comment]
need to rebase to master

File: docs/docs/ProgrammingGuide/keras-support.md [hunk 3]
```diff
@@ -25,9 +25,9 @@ Parameters:
 
 * `def_path` The JSON file path containing the keras model definition to be loaded.
```
Comment (zhichao-li) [inline docs/docs/ProgrammingGuide/keras-support.md]: json_path and hdf5_path?

File: pyspark/bigdl/examples/keras/mnist_cnn.py [hunk 4]
```diff
@@ -48,38 +56,27 @@ def build_keras_model():
...
-    model_json = keras_model.to_json()
-    with open(path, "w") as json_file:
-        json_file.write(model_json)
-
-
 if __name__ == "__main__":
     keras_model = build_keras_model()
     def_path = "/tmp/lenet.json"
-    save_keras_model(keras_model, def_path)
+    save_keras_definition(keras_model, def_path)
```
Comment (zhichao-li) [inline pyspark/bigdl/examples/keras/mnist_cnn.py]: we should give an example that saving definition and also parameters.

Comment (hkvision) [reply pyspark/bigdl/examples/keras/mnist_cnn.py]: imdb example save definition and parameters

**Comment by zhichao-li** [comment]
LGTM

**Comment by hkvision** [comment]
jenkins passed http://172.168.2.101:8080/view/PR-V/job/BigDL-PR-V/603/

**Pull request merged**